src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2604,9 +2604,19 @@ public override BoundNode VisitArrayLength(BoundArrayLength node)
 
         public override BoundNode VisitConditionalGoto(BoundConditionalGoto node)
         {
-            VisitRvalue(node.Condition);
-            Debug.Assert(!this.IsConditionalState);
-            _pendingBranches.Add(new PendingBranch(node, this.State));
+            VisitCondition(node.Condition);
+            Debug.Assert(this.IsConditionalState);
+            if (node.JumpIfTrue)
+            {
+                _pendingBranches.Add(new PendingBranch(node, this.StateWhenTrue));
+                this.SetState(this.StateWhenFalse);
+            }
+            else
+            {
+                _pendingBranches.Add(new PendingBranch(node, this.StateWhenFalse));
+                this.SetState(this.StateWhenTrue);
+            }
```

Comment:
Consider choosing state only based on `node.JumpIfTrue`. Same comment for VB.
```
var state = node.JumpIfTrue ? this.StateWhenTrue : this.StateWhenFalse;
_pendingBranches.Add(new PendingBranch(node, state));
this.SetState(state);
```

Reply:
No, I think that would not be correct. The jump state and the new current state are supposed to be opposites (one for when the branch is taken, and one for when it is not taken).